lib/src/components/board/board.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:viam_sdk/src/resource/base.dart';
+
+import '../../gen/common/v1/common.pb.dart' as common;
+import '../../robot/client.dart';
+
+class BoardStatus {
+  final Map<String, int> analogs;
+  final Map<String, int> digitalInterrupts;
+
+  BoardStatus(this.analogs, this.digitalInterrupts);
+
+  factory BoardStatus.fromProto(common.BoardStatus pbBoardStatus) {
+    BoardStatus boardStatus = BoardStatus(Map<String, int>(), Map<String, int>());
+    pbBoardStatus.analogs.forEach((key, value) => boardStatus.analogs[key] = value.value);
+    pbBoardStatus.digitalInterrupts.forEach((key, value) => boardStatus.digitalInterrupts[key] = value.value.toInt());
```

Comment:
in the TS SDK these values are mapped in the [client](https://github.com/viamrobotics/viam-typescript-sdk/blob/main/src/components/board/client.ts#L63-L71), I thought it was a little cleaner to do it here in the BoardStatus factory constructor. open to suggestions though.